AdvancedHero.cpp

```diff
@@ -0,0 +1,36 @@
+#include "AdvancedHero.h"
+#include <cmath>
+
+
+int AdvancedHero::xpHatar = 100;
+
+AdvancedHero::AdvancedHero(const std::string& name, int hp, const int dmg, const double as) : Hero(name, hp, dmg, as), maxHp(hp){
+}
+
+AdvancedHero AdvancedHero::parseUnit(const std::string& filename){
+    Hero Unit = Hero::parseUnit(filename);
+    return AdvancedHero(Unit.getName(), Unit.getHP(), Unit.getDmg(), Unit.getAttackSpeed());
+}
+
+void AdvancedHero::advancedDamage(AdvancedHero* enemy){
+    xp += std::min(dmg, enemy->getHP());
+    this->damaging(enemy);
+    levelup(this->xp / xpHatar);
+}
+
+void AdvancedHero::levelup(int levelcount){
+    for(int i=0; i<levelcount; i++){
+        maxHp = (int)round((double)maxHp * 1.1);
+        dmg = (int)round((double)dmg * 1.1);
+        hp = maxHp;
+        lvl++;
+        xp -= xpHatar;
```

Comment:
A merge-elt változatban szintlépéskor az attackspeednek is fejlődnie kell, lásd a [feladatleírás](https://github.com/orgs/Teaching-projects/teams/sze-mosze-2020/discussions/9) Továbbiak részét:

>amikor a levelup es attackspeed fejelsztesek talalkoznak, akkor szintlepesnel a cooldown csokkenjen 10%-kal

Reply:
@oliverosz Jogos, nem figyeltem rá. Hozzáadtam ezt is!